torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -203,16 +203,13 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
   }
 
   // Subclasses must override this method to return the backend name
-  virtual const std::string getBackendName() const = 0;
+  virtual const std::string getBackendName() const {
+    TORCH_INTERNAL_ASSERT(false, "getBackendName is not implemented.");
+  };
 
-  virtual c10::intrusive_ptr<ProcessGroup::Work> broadcast(
-      std::vector<at::Tensor>& /* tensors */,
-      const BroadcastOptions& /* opts */ = BroadcastOptions()) {
-    TORCH_CHECK(
-        false,
-        c10::str(
-            "ProcessGroup ", getBackendName(), "does not support broadcast"));
-  }
+  c10::intrusive_ptr<ProcessGroup::Work> broadcast(
```

Comment:
This method needs to be virtual otherwise there might be issues in terms of appropriate dispatching when derived classes override this method.

Reply:
I don't think we want the derived classes to override this method. This is the key entry point to the dispatcher. Will it make more sense to separate the dispatch passable cc ops into a separate file and then route the pybind and any C++ distributed runtime, for instance, reducer to call into the new cc ops?

But then we need a way to make sure people don't accidentally use the old ops. So that's why I think it makes more senses to directly modify the existing ops here.